client/components/domains/connect-domain-step/connect-domain-step-suggested-start.jsx

```diff
@@ -26,16 +25,18 @@ export default function ConnectDomainStepSuggestedStart( {
 	onNextStep,
 	progressStepList,
 	setPage,
+	domainSetupInfo,
 } ) {
 	const { __ } = useI18n();
+	const { data } = domainSetupInfo;
 	const selectedSite = useSelector( getSelectedSite );
 	const goToDnsRecordsPage = () => page( domainManagementDns( selectedSite?.slug, domain ) );
-	const firstStep = isSubdomain( domain )
+	const firstStep = data?.is_subdomain
```

Comment:
When would `data` be undefined?

Reply:
Technically never, since the state is initialized with the default object

```
const [ domainSetupInfo, setDomainSetupInfo ] = useState( defaultDomainSetupInfo );
```

In cases like this I lean towards generous with safety checks like ?., since it has no impact on the logic or performance, but if it was implemented differently, not having the ?. could throw an error.

Would you rather it be a more strict check?